recipes/userspace-rcu/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class UserspaceRCUConan(ConanFile):
+    name = "userspace-rcu"
+    homepage ="https://liburcu.org/"
+    description = "Userspace RCU (read-copy-update) library"
+    topics = ("urcu")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-2.1"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "model": ["generic", "mb", "signal", "bp"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "model": "generic",
+    }
+    build_requires = (
+        "libtool/2.4.6",
+    )
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            raise ConanInvalidConfiguration("Only Linux/FreeBSD supported")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            self.run("./bootstrap")
+            tools.mkdir("_build")
+            with tools.chdir("_build"):
+                env_build = AutoToolsBuildEnvironment(self)
+                extra_args = list()
+                if self.options.shared:
+                    extra_args.append('--enable-static=no')
+                else:
+                    extra_args.append('--enable-shared=no')
+                env_build.configure("../", args=extra_args, build=False, host=False, target=False)
+                env_build.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE*", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", excludes=("*_build/*", "*doc/*", "*tests/*"), src="{}/src".format(self._source_subfolder), dst="include/", keep_path=True)
+        self.copy("*.h", excludes=("*_build/*", "*doc/*", "*tests/*"), src="{}/include".format(self._source_subfolder), dst="include/", keep_path=True)
+        self.copy("*.h", src="{}/_build/include/urcu/".format(self._source_subfolder), dst="include/urcu", keep_path=False)
+        lib_name = ''
+        if self.options.model == "generic":
+            lib_name = 'liburcu'
+        elif self.options.model == "mb":
+            lib_name = 'liburcu-mb'
+        elif self.options.model == "signal":
+            lib_name = 'liburcu-signal'
+        else:
+            lib_name = 'liburcu-bp'
```

Comment:
I would have preferred something like the following, that resembles the same logic you implemented in the `package_info` method:

```suggestion
        lib_name = 'liburcu'
        if self.options.model == "mb":
            lib_name += '-mb'
        elif self.options.model == "signal":
            lib_name += '-signal'
        elif self.options.model == "bp":
            lib_name += '-bp'
```